compiler/daml-lf-ast/src/DA/Daml/LF/Ast/Type.hs

```diff
@@ -104,6 +105,7 @@ substitute subst = go (Map.foldl' (\acc t -> acc `Set.union` freeVars t) Set.emp
             in  TForall (v1, k) (go fvSubst1 subst1 t)
         | otherwise -> TForall (v0, k) (go fvSubst0 (Map.delete v0 subst0) t)
       TTuple fs -> TTuple (map (second go0) fs)
+      TNat n -> TNat n
```

Comment:
Think about the extra allocation here with `-O0`! :stuck_out_tongue:

Reply:
oh, how awful!